packages/Heading/Heading.jsx

```diff
@@ -8,60 +8,40 @@ import HeadingSup from './HeadingSup/HeadingSup'
 
 import styles from './Heading.modules.scss'
 
-const BaseHeading = ({ level, color, size, children, ...rest }) =>
-  React.createElement(
-    level,
-    {
-      ...safeRest(rest),
-      className: joinClassNames(size, color),
-    },
-    children
-  )
-
-BaseHeading.propTypes = {
-  level: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']).isRequired,
-  color: PropTypes.string.isRequired,
-  size: PropTypes.string.isRequired,
-  children: PropTypes.node.isRequired,
-}
-
 /**
  * Page headings. Renders an HTML `<h1-h4>` element.
  *
  * @version ./package.json
  */
-const Heading = ({ level, invert, children, ...rest }) => {
-  const baseHeadingProps = {
-    ...rest,
-    level,
+const Heading = ({ level, tag = level, invert, children, ...rest }) => {
+  const getClassNames = () => {
```

Comment:
Let's prefer to extracting utility methods from the functional component, similar to Button: https://github.com/telus/tds-core/blob/7485396b857a2c2c2baa657abacacfd9123bba5d/packages/Button/Button.jsx#L9-L15